core/drivers/zynqmp_pm.c

```diff
@@ -121,6 +121,9 @@ static TEE_Result efuse_op(enum efuse_op op, uint8_t *buf, size_t buf_sz,
 		goto out;
 	}
 
+	if (op == EFUSE_WRITE)
+		memcpy(tmpbuf, buf, buf_sz);
```

Comment:
It is used for both.

For `EFUSE_WRITE` it is filled with data to be programmed just before entering the PMU MicroBlaze programming operation. (as we need to know what to write).

Thus we need to make sure that caches are written in real memory before the call.

For `EFUSE_READ` we need to have buffer for which PMU MicroBlaze will write the data read from eFuses.

And in here we need in invalidate caches after the call so that we go to read from real memory.